### PR TITLE
create new helm chart for deploying RTCD

### PR DESCRIPTION
**Pull request by angeloskyratzakos**
#### Summary
Create a new helm chart for deploying RTCD service

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43735

### CODE REVIEW

File: charts/mattermost-rtcd/values.yaml
Comment (stafot) [inline charts/mattermost-rtcd/values.yaml:47]: I think we also need `className`.

Comment (angeloskyratzakos) [reply charts/mattermost-rtcd/values.yaml:47]: Thanks @stafot I have updated the whole ingress section with the newer API version

**Review by stafot — APPROVED** [review]
LGTM

File: charts/mattermost-rtcd/Chart.yaml
Comment (stylianosrigas) [inline charts/mattermost-rtcd/Chart.yaml:4]: Please add some more info. such as Maintainers. You can see example here -> https://github.com/mattermost/mattermost-helm/blob/master/charts/mattermost-enterprise-edition/Chart.yaml

File: charts/mattermost-rtcd/values.yaml
Comment (stylianosrigas) [inline charts/mattermost-rtcd/values.yaml:55]: lets update this to ingressClassName

**Review by phoinixgrr — COMMENTED** [review]
Awesome work there 👍  🚀 

Ct lint approves 
```
❯ ct lint --charts charts/mattermost-rtcd --validate-yaml
Linting charts...
Version increment checking disabled.

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 mattermost-rtcd => (version: "1.0.0", path: "charts/mattermost-rtcd")
------------------------------------------------------------------------------------------------------------------------

Linting chart 'mattermost-rtcd => (version: "1.0.0", path: "charts/mattermost-rtcd")'
Validating /Users/phoinix/Repos_Public/mattermost-helm/charts/mattermost-rtcd/Chart.yaml...
Validation success! 👍
Validating maintainers...
==> Linting charts/mattermost-rtcd
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ mattermost-rtcd => (version: "1.0.0", path: "charts/mattermost-rtcd")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

Left two minor comments 



Morover tests seem to fail. Should they be fixed before merging ?

**Review by phoinixgrr — APPROVED** [review]
:pasok:

**Comment by LvLs9** [comment]
Hi! Am I right that I need RTCD if I want to use new beta-calls?

**Comment by heyplusyou** [comment]
Seems so, mattermost server does not have rtc udp port 8443, only this has. Could someone confirm setup?

**Comment by onpaws** [comment]
Is this chart supposed to be supported for self-hosting? If anyone has got it working would would appreciate your thoughts.

Been trying to follow [these steps](https://docs.mattermost.com/configure/calls-deployment.html) but no luck yet. 

The good news is
`helm upgrade mattermost-rtcd mattermost/mattermost-rtcd -f mattermost-rtcd-values.yaml --namespace mattermost-rtcd --create-namespace --install --debug`
seemed to finish (🎉 ). 
The not so good news is calls are still ending after 10s, like they did before. I suspect UDP still isn't exposed because Mattermost client console log says:
```
com.mattermost.calls: rtc closed
com.mattermost.calls_20b1f0b7d4b3b754_bundle.js:2 com.mattermost.calls: disconnect
```

Where does [this Ingress Class](https://github.com/mattermost/mattermost-helm/blob/6091ddd2e1283a8a1c2c49f0aab2ef0d5d349357/charts/mattermost-rtcd/values.yaml#L65) come from?
image
Should it not be just `nginx`? (cluster has Nginx Ingress installed).

Meanwhile the service AFAICT looks good. 
image

So yeah, external traffic is apparently not hitting UDP port. Not sure where to look next 🤔 
If anyone has any suggestions would be appreciated 🙏


---
Chart version: 1.1.1
mattermost-rtcd version: 0.7.0
Kubernetes: 1.24.4
Ingress: Nginx Ingress 

values.yaml
```yaml
image:
  repository: mattermost/rtcd
  tag: "v0.7.0"

ingress:
  enabled: true
  hosts:
    - host: mattermost-rtcd.mydomain.dev
      paths:
        - "/"
```

**Pull request merged**
